### PR TITLE
[KED-923] Add YAMLLocalDataset in `contrib`

### PR DESCRIPTION
**Pull request by Minyus**
## Notice

- [x ] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

## Motivation and Context
YAML file format is not supported by kedro.io.
See #64 for details.
This PR replaces https://github.com/quantumblacklabs/kedro/pull/66

## How has this been tested?
Test code was added and run.

## Checklist

- [x ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x ] Updated the documentation to reflect the code changes
- [x ] Added new entries to the `RELEASE.md` file
- [x ] Added tests to cover my changes
- [ ] Assigned myself to the PR

### CODE REVIEW

File: kedro/contrib/io/yaml_local/yaml_local.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            version: If specified, should be an instance of
+                ``kedro.io.core.Version``. If its ``load`` attribute is
+                None, the latest version will be loaded. If its ``save``
+                attribute is None, save version will be autogenerated.
+
+        """
+        super().__init__(Path(filepath), version)
+
+        # Handle default load and save arguments
+        self._save_args = copy.deepcopy(self.DEFAULT_SAVE_ARGS)
```
Comment (Flid) [inline kedro/contrib/io/yaml_local/yaml_local.py]: Please look at the previous comments https://github.com/quantumblacklabs/kedro/pull/66/files/98aad7acf04193d85d60012a788302772fc540a7#r311487946
We haven't done this for the core data sources, but still please keep the contrib package consistent.

Comment (Minyus) [reply kedro/contrib/io/yaml_local/yaml_local.py]: OK.

**Comment by Minyus** [comment]
@Flid @lorenabalan
OK. I modified the code as suggested/requested using `DefaultArgumentsMixIn` and passed all the checks. Could you please review it again?

File: kedro/contrib/io/yaml_local/__init__.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""``YAMLLocalDataset`` is an ``AbstractDataSet`` to save and load YAML files.
```
Comment (lorenabalan) [inline kedro/contrib/io/yaml_local/__init__.py]:
```suggestion
"""``YAMLLocalDataset`` is an ``AbstractVersionedDataSet`` to save and load YAML files.
```

Comment (Minyus) [reply kedro/contrib/io/yaml_local/__init__.py]: OK. I corrected it.

**Pull request merged**